crates/eigen-cli/src/bls.rs

```diff
@@ -0,0 +1,60 @@
+use crate::args::BlsKeystoreType;
+use crate::EigenBlsKeyStoreError;
+use rust_bls_bn254::keystores::{pbkdf2_keystore::Pbkdf2Keystore, scrypt_keystore::ScryptKeystore};
+
+/// BlsKeystore types  
+pub enum BlsKeystore {
+    Pbkdf2,
+    Scrypt,
+}
+
+impl BlsKeystore {
+    /// Create a new [`BlsKeystore`] instance.
+    /// [`BlsKeystore::Pbkdft`] or [`BlsKeystore::Scrypt`]
+    pub fn new_keystore(
+        self,
+        secret_key: String,
+        output_path: String,
+        password: Option<&str>,
+    ) -> Result<(), EigenBlsKeyStoreError> {
+        match self {
+            BlsKeystore::Pbkdf2 => {
```

Comment:
is there a common trait for keystores? if so you could make the keystore part of the struct at build time and thus avoid to have this match (not as problematic as the other match below if it's hard though)

Reply:
umm, i think  it will become more complex